### ISSUE TITLE
Bokeh Images not showing on IE 11

### ISSUE THREAD

**Issue by hervo63** [bokeh/bokeh#6976]
All  bokeh images are not displayed on internet explorer
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
All Bokeh Version (11.0 to 12.9)
Windows 7, internet explorer 11.0.96
#### Description of expected behavior and the observed behavior
Images are not displayed on Internet explorer, 
using the demo, bokeh server or standalone HTML

#### Complete, minimal, self-contained example code that reproduces the issue
http://bokeh.pydata.org/en/latest/docs/gallery/image_rgba.html
working on Firefox bot not IE11

#### Stack traceback and/or browser JavaScript console output
```
HTML1300: Navigation occurred.
File: image_rgba.html
Bokeh: BokehJS not loaded, scheduling load and callback at Wed Sep 27 2017 17:07:48 GMT+0200 (W. Europe Daylight Time)
   "Bokeh: BokehJS not loaded, scheduling load and callback at"
   [date] Wed Sep 27 2017 17:07:48 GMT+0200 (W. Europe Daylight Time)

Bokeh: injecting script tag for BokehJS library:  https://cdn.pydata.org/bokeh/release/bokeh-0.12.9.min.js
Bokeh: injecting script tag for BokehJS library:  https://cdn.pydata.org/bokeh/release/bokeh-widgets-0.12.9.min.js
Bokeh: injecting script tag for BokehJS library:  https://cdn.pydata.org/bokeh/release/bokeh-tables-0.12.9.min.js
Bokeh: injecting script tag for BokehJS library:  https://cdn.pydata.org/bokeh/release/bokeh-gl-0.12.9.min.js
Bokeh: all BokehJS libraries loaded
Bokeh: BokehJS plotting callback run at Wed Sep 27 2017 17:07:48 GMT+0200 (W. Europe Daylight Time)
   "Bokeh: BokehJS plotting callback run at"
   [date] Wed Sep 27 2017 17:07:48 GMT+0200 (W. Europe Daylight Time)

[bokeh] setting log level to: 'info'
[bokeh] JS/Python version mismatch
[bokeh] Library versions: JS (0.12.9)  /  Python (0.12.8rc2)
[bokeh] Will inject Bokeh script tag with params {"docid":"c0833d93-3593-4b74-8480-a10ec92c2aa3","elementid":"e5f54751-9ed1-47a2-9bc8-63c8993ccdf9","modelid":"2c106691-03ca-435a-9c65-f0347b48527c"}
Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-0.12.9.min.css
Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-0.12.9.min.css
Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-0.12.9.min.css
Bokeh: all callbacks have finished
```

#### Screenshots or screencasts of the bug in action
![issue bokeh](https://user-images.githubusercontent.com/13102254/30921389-b392fce8-a3a6-11e7-8c72-7964b9a2bf5c.png)

**Comment by bryevdv** [comment]
cc @mattpap in case you have any quick ideas

**Comment by blacknapkins** [comment]
Hi,
I built and ran the example above with no problems found.
http://bokeh.pydata.org/en/latest/docs/gallery/image_rgba.html

With the following configuration:
Windows 7
IE 11.0.9600.18792
Bokeh 0.12.6
Bokeh 0.12.7 (The last formal version available via anaconda navigator)
Bokeh 0.12.10dev2 (The latest dev build)
Bokeh in standalone mode.
Python 3.5.3

I include the trace below

```
DOM7011: The code on this page disabled back and forward caching. For more information, see: http://go.microsoft.com/fwlink/?LinkID=291337
File: image_rgba.html
HTML1300: Navigation occurred.
File: image_rgba.html
[bokeh] setting log level to: 'debug'
[bokeh] embed: computed ws url: ws:///C:/Data/scoot_stuff/experiments_in_python/Bokeh_exp/Bokeh_exp/image_rgba.html/ws
[bokeh] Creating Document from JSON
[bokeh] Library versions: JS (0.12.10dev2)  /  Python (0.12.10dev2)
[bokeh] Toolbar: activating tool: PanTool (1048) for event type 'pan'
[bokeh] PlotCanvas initialized
[bokeh] Adding root: Plot(1031)
[bokeh] PlotCanvas attached to document
[bokeh] CanvasView initialized
[bokeh] ImageRGBA GlyphRenderer (1059): set_data finished in 0ms
[bokeh] Button tool: SaveTool
[bokeh] Button tool: ResetTool
[bokeh] Button tool: HelpTool
[bokeh] PlotView initialized
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 0ms
[bokeh] initial ranges set
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 0ms
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 0ms
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 1ms
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 0ms
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 0ms
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 0ms
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 0ms
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 1ms
[bokeh] Rendering CanvasView with width: 600, height: 600, pixel ratio: 1
[bokeh] ImageRGBA GlyphRenderer (1059): render finished in 0ms
```
If I can help further with this please let me know.

**Comment by bryevdv** [comment]
@blacknapkins Just to be clear you are saying you could *not* reproduce any problems? (i.e. the examples were working as expected?) For reference, on  a Win7 VM with IE11 I am seeing no issues on the website for the ImageRGBA example:

<img width="774" alt="screen shot 2017-11-07 at 14 04 13" src="https://user-images.githubusercontent.com/1078448/32515082-8f579e90-c3c4-11e7-8c0e-9bf057312e5e.png">

@hervo63 we would need to try an collect additional information that might help lead us to tracking any real problem down. I am going to close this for now since it cannot be reproduced. However, if you find more details or information that will point to reproducing this, please re-open.

**Comment by blacknapkins** [comment]
@bryevdv correct. I could not reproduce this issue with the configurations that I stated.

**Comment by hervo63** [comment]
@volkerjaenisch : I think you fixed this issue, could you give some hints on how you did it?
Thanks,
Max

**Comment by volkerjaenisch** [comment]
@hervo63 I currently have no win7 at hand. With a Win10 IE11 it runs flawlessly. Since bokeh does not give you a trace it is not likely a bokey error. I suspect the error may lie in your PC e.g. the graphics driver.

**Comment by bbartoldson** [comment]
@hervo63 @bryevdv With Linux and Firefox, I am unable to get a page to display for image_rgba.py, and there are no errors from Bokeh. I began investigating image_rgba.py after I switched from 1-channel images to 4-channel images (going from figure.image() to figure.image_rgba()) in my own code, and Bokeh stopped displaying a page--with no errors. 

My problem was solved after I realized I had accidentally changed my numpy image arrays to float type when adding the extra channels. Changing the dtype to np.uint8 led to a page being displayed. The image_rgba example code is already using an int dtype, so I don't know if the issues are related, but I wanted to post this in case it comes up in the future.
